spyder/plugins/explorer/widgets/explorer.py

```diff
@@ -405,24 +391,43 @@ def get_file_associations(self, fname):
     @Slot()
     def edit_filter(self):
         """Edit name filters."""
-        dialog = QInputDialogMultiline(
-            self, _('Edit filename filters'),
-            _('Name filters:'),
-            ", ".join(self.name_filters))
+        # Create Dialog
+        dialog = QDialog(self)
         dialog.resize(500, 300)
-        result = dialog.exec_()
-        if result:
-            filters = dialog.text_edit.toPlainText()
-            filters = [f.strip() for f in to_text_string(filters).split(',')]
-            self.parent_widget.sig_option_changed.emit('name_filters', filters)
-            self.set_name_filters(filters)
-
-    @Slot(bool)
-    def toggle_all(self, checked):
-        """Toggle all files mode"""
-        self.parent_widget.sig_option_changed.emit('show_all', checked)
-        self.show_all = checked
-        self.set_show_all(checked)
+        dialog.setWindowTitle(_('Edit filter settings'))
+
+        # Create dialog contents
+        description_label = QLabel(
+            _('Filter files by name, extension, or more using glob patterns. '
+              'Please enter the glob patterns, separated by commas.'))

```

Comment:
```suggestion
            _('Filter files by name, extension, or more using glob patterns. '
              'Please enter the glob patterns, separated by commas, '
              at least one of which a file must match to be shown when '
              filters are enabled.'))
```

To address what @juanis2112 brought up and be a little more clear, I propose the above as a suggestion for the dialog text.